### PR TITLE
New Homepage + Navigation [Feature]

### LINKED ISSUE #645: New home page + nav
Figma designs are [here](https://www.figma.com/file/dvX3yg7FMGJSWFaqHWrWgQ/Justice-Counts-Publisher?type=design&node-id=8312-20721&t=6kk2me022NS2AtJw-0). You will likely have questions on desired behavior and logic, so please do coordinate with Jessica and Forrest as needed! Also feel free to close this task and break it up into smaller ones :)

### PR DESCRIPTION
**Pull request by mxosman**
## Description of the change

Implements newly designed navigation and updates current routing & links accordingly.

Figma: https://www.figma.com/file/dvX3yg7FMGJSWFaqHWrWgQ/Justice-Counts-Publisher?type=design&node-id=8312%3A20722&t=WfYvIkrCN200ayJa-1

Demo: https://mahmoudtest---justice-counts-web-qqec6jbn6a-uc.a.run.app/

*This PR also serves as the main PR for this feature that will be merged in once the feature is ready to be launched. It is comprised of the following PRs below.

| PR  | Description |
|-|-|
#662 | New Navigation: Routing & UI Scaffolding |
#682 | New Homepage: UI Scaffolding & Task Card Interactions [1/n] |
[#21610 ](https://github.com/Recidiviz/recidiviz-data/pull/21610) | New Homepage: Implement new endpoint to support new homepage |
#730 | New Homepage: Support agencies with multiple systems [2/n] |
#731 | Agency Settings & Metric Config: Action Required treatment |
#740 | New Homepage: Scroll to metric in Data Entry Form after clicking on "Manual Entry" action link in task card [3/n] |
#741 | New Homepage: Add "Home" button to Data Entry Form & Publish success modal [4/n] |
#745 | New Homepage: Link “Manual Entry” action link to Create Report page for users who do not have a records [5/n] |
#751 | New Homepage: Bug Fixes from Pre-playtesting [6/n] |
#752 | New Homepage: P0 Mobile Responsiveness [7/n] |
#759 | New Homepage: Playtesting Followups (P0) |
#762 | New Homepage: Playtesting Followups (P1) |
#764 | New Homepage: Deprecate all old Onboarding and Guidance |
#765 | New Homepage: Playtesting Followup - add helpful message to Create Records page when redirected from task card |
#768 | New Homepage: Toast Adjustment |
#785 | New Homepage: CSG Playtesting Followups |
... | ... |

## Related issues

Closes #645 #667 #671

### Development

**This box MUST be checked by the submitter prior to merging**:
- [x] **Double- and triple-checked that there is no Personally Identifiable Information (PII) being mistakenly added in this pull request**

These boxes should be checked by the submitter prior to merging:
- [ ] Tests have been written to cover the code changed/added as part of this pull request

### Code review

These boxes should be checked by reviewers prior to merging:

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [ ] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

File: common/components/Button/Button.tsx [hunk 1]
```diff
@@ -63,7 +61,6 @@ export function Button({
         borderColor={borderColor}
         size={size}
         disabled={disabled}
-        enabledDuringOnboarding={enabledDuringOnboarding}
```
Comment (mxosman) [inline common/components/Button/Button.tsx:66]: Slowly deprecating the guidance items that are getting in the way.

File: common/components/Dropdown/Dropdown.styled.tsx [hunk 2]
```diff
@@ -149,3 +161,17 @@ export const CustomDropdownMenuItem = styled(DropdownMenuItem)<{
...
+  height: 100%;
+  width: 100%;
+  display: flex;
+  justify-content: ${({ highlightIcon }) =>
+    highlightIcon ? "space-between" : "flex-start"};
+  align-items: center;
+  gap: 16px;
+  color: inherit !important;
+  ${({ highlightIcon }) => highlightIcon && "padding-left: 8px;"}
+`;
```
Comment (mxosman) [inline common/components/Dropdown/Dropdown.styled.tsx:177]: Updating shared dropdown to new design.

File: common/components/Dropdown/Dropdown.tsx [hunk 3]
```diff
@@ -104,16 +106,23 @@ export function Dropdown({
...
                   noHover={noHover}
+                  highlight={highlight && !highlightIcon}
                 >
-                  {optionLabel}
+                  <Styled.OptionLabelWrapper
+                    highlightIcon={Boolean(highlightIcon)}
+                  >
+                    {!highlightIcon && icon}
+                    {optionLabel}
+                    {highlightIcon && highlight && highlightIcon}
```
Comment (mxosman) [inline common/components/Dropdown/Dropdown.tsx:124]: This renders a regular icon (if provided) on the left side of the menu label (e.g. menu under your profile) and a highlight icon to the right side of the menu label (if the menu item is highlighted - e.g. agencies dropdown menu).

File: publisher/src/App.tsx [hunk 4]
```diff
@@ -58,26 +57,14 @@ const App: React.FC = (): ReactElement => {
...
             <Route path="/agency/:agencyId/*" element={<Router />} />
             <Route
               path="*"
               element={
                 <Navigate
-                  to={`/agency/${initialAgency}/${
-                    hasCompletedOnboarding
-                      ? REPORTS_LOWERCASE
-                      : "getting-started"
-                  }`}
```
Comment (mxosman) [inline publisher/src/App.tsx:80]: Removing `hasCompletedOnboarding` from this file.

File: publisher/src/components/Menu/Menu.tsx [hunk 5]
```diff
@@ -206,177 +179,108 @@ const Menu: React.FC = () => {
...
+                  onClick={() => {
+                    navigate(`settings/${path}`);
+                    handleCloseMobileMenu();
+                  }}
+                >
+                  {displayLabel}
+                </SubMenuItem>
+              ))}
+            </SubMenuContainer>
+          )}
```
Comment (mxosman) [inline publisher/src/components/Menu/Menu.tsx:265]: No changes here.

File: publisher/src/utils/helperUtils.ts [hunk 6]
```diff
@@ -234,7 +234,7 @@ export function memoizeDebounce<F extends (...args: any[]) => any>(
  * @param location
  */
 export function removeAgencyFromPath(location: string) {
-  const agencyRegex = /\/agency\/\d+\//i;
+  const agencyRegex = /\/agency\/\d+\/?/i;
```
Comment (mxosman) [inline publisher/src/utils/helperUtils.ts:237]: Making the `/` forward slash character at the end of the path optional. That way `/agency/148` or `/agency/148/` can be removed. Currently, something like `/agency/148` without the forward slash is not replaced.

File: publisher/src/components/DataUpload/UploadFile.tsx [hunk 7]
```diff
@@ -141,7 +141,9 @@ export const UploadFile: React.FC<UploadFileProps> = ({
       </Instructions>
 
       <DragDropContainer ref={dragDropAreaRef} dragging={dragging}>
-        <DragDropIcon src={fileIcon} alt="" />
+        <DragDropIconWrapper>
+          <FileIcon />
```
Comment (mxosman) [inline publisher/src/components/DataUpload/UploadFile.tsx:145]: It's more optimal to render svgs as react components via the pattern here so they load instantly vs. loading an svg via an `` tag which takes a super quick but noticeable amount of time to render.

**Comment by mxosman** [comment]
I can't think of a good way to gate this (open to ideas!) - but I am happy to leave this as the parent branch and continue stacking until we're ready to release this feature.

File: common/components/Dropdown/Dropdown.styled.tsx (hunk 2 above)
Comment (lilidworkin) [reply common/components/Dropdown/Dropdown.styled.tsx:177]: Nice! Love that we can make use of the shared components

File: publisher/src/components/DataEntryInterstitial/DataEntryInterstitial.tsx [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+
+import React from "react";
+import { useNavigate } from "react-router-dom";
+
+import { ReactComponent as ManualEntryIcon } from "../assets/manual-entry-icon.svg";
+import { ReactComponent as UploadFolderIcon } from "../assets/upload-folder-icon.svg";
+import { REPORTS_LOWERCASE } from "../Global/constants";
+import * as Styled from ".";
+
+export const DataEntryInterstitial = () => {
```
Comment (lilidworkin) [inline publisher/src/components/DataEntryInterstitial/DataEntryInterstitial.tsx:26]: This page looks great!

File: publisher/src/components/DataUpload/UploadFile.tsx (hunk 7 above)
Comment (lilidworkin) [reply publisher/src/components/DataUpload/UploadFile.tsx:145]: Cool!

**Review by lilidworkin — COMMENTED** [review]
@mxosman this looks awesome! Just one request re: the staging badge.

How does this interact with first-run user experience btw? I keep forgetting we have that. I wonder if we need to get rid of it at some point if it will get too outdated. Might be worth taking a video of it and showing it to Jessica + Forrest to see if they want to do something with it

File: common/components/Dropdown/Dropdown.tsx (hunk 3 above)
Comment (mxosman) [reply common/components/Dropdown/Dropdown.tsx:124]: > SUPER nit but it's redundant to have two highlightIcons in this && statement, right? I mean I think I see the intention - if hightlightIcon is not null and we're supposed to hightlight, then show the hightlight Icon, but I think highlight && highlightIcon will do the same thing?

You're totally right! Will update! Thanks for catching!



> Why do we also need `highlightIcon={Boolean(highlightIcon)}`?

This is to add some different spacing to the `OptionLabelWrapper`.

> Might be good to clarify in a comment the difference between highlightIcon and icon

Great idea! Will do - this is definitely not clear!

File: publisher/src/components/Header/Header.tsx [hunk 9]
```diff
@@ -15,49 +15,26 @@
...
   const defaultAgency = userStore.getInitialAgencyId();
   const onLogoClick = () =>
-    hasCompletedOnboarding
-      ? navigate(
-          `/agency/${
-            isAgencyValid ? agencyId : defaultAgency
-          }/${REPORTS_LOWERCASE}`
-        )
-      : navigate(guidancePaths.home);
-  const badge = () => {
```
[the remark this answers is not in the document]
Comment (mxosman) [reply publisher/src/components/Header/Header.tsx:45]: Oh yeah! I knew I forgot to do something. Sounds good!

File: publisher/src/components/Menu/Menu.tsx [hunk 10]
```diff
@@ -29,67 +28,58 @@ import { useLocation, useNavigate, useParams } from "react-router-dom";
...
   const location = useLocation();
   const windowWidth = useWindowWidth();
+
   const [settingsSearchParams, setSettingsSearchParams] =
     useSettingsSearchParams();
   const [isMobileMenuOpen, setIsMobileMenuOpen] = useState(false);
 
   const pathWithoutAgency = removeAgencyFromPath(location.pathname);
   const currentAgency = userStore.getAgency(agencyId);
-  const systemMetricKey = getActiveSystemMetricKey(settingsSearchParams);
```
[the remark this answers is not in the document]
Comment (mxosman) [reply publisher/src/components/Menu/Menu.tsx:77]: Should be all safe to remove. This was all only relevant to the previous Guidance and its metric config completion status toast.

**Comment by mxosman** [comment]
Dev environment badges are back!

Screenshot 2023-05-31 at 4 49 16 PM
Screenshot 2023-05-31 at 4 49 09 PM

File: common/components/Dropdown/Dropdown.tsx (hunk 3 above)
Comment (mxosman) [reply common/components/Dropdown/Dropdown.tsx]: [b5ece89](https://github.com/Recidiviz/justice-counts/pull/662/commits/b5ece89e4b13a93b6efb0892afc6fd059a0690b4)

**Review by lilidworkin — APPROVED** [review]
Fantastic, these changes look great @mxosman !

Since this branch will be unmerged for a while, set a reminder to yourself to rebase it on main every day or two so.

**Comment by mxosman** [comment]
Here goes nothing!

**Pull request merged**